### ISSUE TITLE
chez/chez014 test fails

### ISSUE THREAD

**Issue by kozross** [idris-lang/Idris2#860]
When I build (bootstrap from Chez), I get this:

```
chez/chez014: FAILURE
Expected:
Received: hello world!
Received: echo: hello world!
1/1: Building Echo (Echo.idr)
Main> Main> Bye for now!

Given:
[server] Failed to bind socket with error: 99
1/1: Building Echo (Echo.idr)
Main> Main> Bye for now!
```

**Comment by gallais** [comment]
Are you [using arch](https://aur.archlinux.org/packages/idris2/) by any chance?
It looks like the libs are built out of order leading to this issue.

**Comment by kozross** [comment]
Yes, I am, but this doesn't just affect the AUR package. I got this same error when I cloned from Github and followed the instructions.

**Comment by kozross** [comment]
I've provided what additional info I thought necessary to move this along. Is there more I can provide so that this could be resolved, @gallais ?

**Comment by gallais** [comment]
Ah, I hadn't noticed you were the one who had posted the upstream issue. :sweat_smile: 
I know next to nothing about the network code so it's probably better if
someone more knowledgeable steps in.

**Comment by kozross** [comment]
I can confirm this issue is still present on 0.3.0.

**Comment by ShinKage** [comment]
I don't know much about the network code, but the error is raised when the test code tries to bind a new socket on localhost:0 (which should mean a free port on localhost). Can you check if it is indeed possible to do so in your system?

**Comment by kozross** [comment]
@ShinKage Not a network expert - how would I check this?

**Comment by ShinKage** [comment]
@kozross Sorry, I must have involuntarily skipped the notification. I'm very far from a network expert myself but you can try to directly call the C primitives, simplifying all the checks done in the support library, in particular you can take a look at the two functions called by the test [here](https://github.com/idris-lang/Idris2/blob/master/support/c/idris_net.c#L76) and [here](https://github.com/idris-lang/Idris2/blob/master/support/c/idris_net.c#L129). They will probably fail again, but maybe you can get more helpful error descriptions since the Idris2 support code does not propagate the exact error number.

**Comment by kozross** [comment]
99 is `EADDRNOTAVAIL`. What port is the server trying to start on in the test code?

**Comment by gallais** [comment]
[Looks like `0`](https://github.com/idris-lang/Idris2/blob/master/tests/chez/chez014/Echo.idr#L13)

**Comment by kozross** [comment]
Binding to port 0 is not allowed, I'm pretty sure.

**Comment by gallais** [comment]
From what I gather `0` means "just give me any port you like".

**Comment by kozross** [comment]
Either way, this is _extremely_ flaky.
